DemoApp/Sources/Views/Controls/DemoControls.swift

```diff
@@ -0,0 +1,332 @@
+        .onReceive(chatViewModel?.$isChatVisible) { isChatVisible = canOpenChat && $0 }
+        .onReceive(viewModel.$call, perform: { call in
+            reactionsHelper.call = call
+        })
+        .halfSheetIfAvailable(isPresented: $isChatVisible, onDismiss: { chatViewModel?.isChatVisible = false }) {
+            if let chatViewModel = chatViewModel, let channelController = chatViewModel.channelController {
+                VStack {
+                    ChatControlsHeader(viewModel: viewModel)
+                    ChatView(
+                        channelController: channelController,
+                        chatViewModel: chatViewModel,
+                        callViewModel: viewModel
+                    )
+                }
+            }
+        }
+    }
+}
+
+struct ChatControlsHeader: View {
+
+    @Injected(\.streamVideo) var streamVideo
+    @Injected(\.images) var images
+    @Injected(\.colors) var colors
+
+    @Environment(\.chatVideoViewModel) var chatViewModel
+
+    private let size: CGFloat = 50
+
+    @ObservedObject var viewModel: CallViewModel
+
+    init(viewModel: CallViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        AppControlsWithChat(viewModel: viewModel, canOpenChat: false)
+    }
+}
+
+struct ChatIconView: View {
+
+    @Injected(\.images) var images
+    @Injected(\.colors) var colors
+
+    @ObservedObject var viewModel: StreamChatVideoViewModel
+    let size: CGFloat
+
+    init(viewModel: StreamChatVideoViewModel, size: CGFloat = 50) {
+        self.viewModel = viewModel
+        self.size = size
+    }
+
+    var body: some View {
+        Button(
+            action: {
+                viewModel.isChatVisible.toggle()
+            },
+            label: {
+                CallIconView(
+                    icon: viewModel.isChatVisible ? .init(systemName: "message.fill") : .init(systemName: "message"),
+                    size: size,
+                    iconStyle: viewModel.isChatVisible ? .primary : .transparent
+                ).overlay(
+                    VStack {
+                        HStack {
+                            Spacer()
+                            if viewModel.unreadCount > 0 {
+                                Text("\(viewModel.unreadCount)")
+                                    .font(.caption.monospacedDigit())
+                                    .foregroundColor(colors.text)
+                                    .padding([.leading, .trailing], 4)
+                                    .padding([.top, .bottom], 2)
+                                    .background(Color.red)
+                                    .clipShape(Capsule())
+                                    .clipped()
+                            }
+                        }
+                        Spacer()
+                    }
+                )
+            }
+        )
+        .accessibility(identifier: "chatToggle")
+    }
+}
+
+struct VideoIconView: View {
```

Comment:
Why is this one not used from the defaults?

Reply:
No reason, just a remnant from the previous repo. Replacing them